docs/docs/rest-api/public/api/v2/apps.raml

```diff
@@ -0,0 +1,451 @@
+    responses:
+      200:
+        body:
+          application/json:
+            example: !include examples/app_list.json
+      404:
+        description: No task found with this `app_id`.
+        body:
+          application/json:
+            example: |
+              { "message": "App '/not_existent' does not exist" }
+
+  ###
+  # Update a specific app
+  #
+  put:
+    description:
+      Replaces parameters of a running application. If no application with the given id
+      exists, it will be created. If there is an application with this id, all running instances get
+      upgraded to the new definition.
+    is: [ secured ]
+    queryParameters:
+      force:
+        description: If there is an already running deployment that affects the
+        type: boolean
+        default: false
+    body:
+      application/json:
+        example: !include examples/app.json
+        schema: AppDefinition
+    responses:
+      200:
+        description:
+          A deployment is started which has a unique deployment identifier.
+          The related deployment can be fetched from the /v2/deployments endpoint.
+          If the deployement is gone from the list of deployments, than this means it is finished.
+          As long as the deployment runs, the effect of that change operation is visible only partially.
+        body:
+          application/json:
+            example: !include examples/deployments_result.json
+      404:
+        description: No task found with this `app_id`.
+        body:
+          application/json:
+            example: |
+              { "message": "App '/not_existent' does not exist" }
+      409:
+        description:
+          There is an already running deployment, that affects this application.
+          To override this deployment, use the force=true flag
+        body:
+          application/json:
+            example: |
+              {
+               "message":"App is locked by one or more deployments. Override with the option '?force=true'. View details at '/v2/deployments/<DEPLOYMENT_ID>'.",
+               "deployments":[{"id":"97c136bf-5a28-4821-9d94-480d9fbb01c8"}]
+              }
+      400:
+        description: The application definition provided in the body is not valid.
+        body:
+          application/json:
+            example: |
+              {"message":"Invalid JSON","details":[{"path":"/id","errors":["error.expected.jsstring"]}]}
+      422:
+        description: The entity send can not be preocessed, since there are validation errors
+        body:
+          application/json:
+            example: |
+              {
+                "message": "Bean is not valid",
+                "errors": [
+                  {
+                    "attribute": "upgradeStrategy.minimumHealthCapacity",
+                    "error": "is greater than 1"
+                  }
+                ]
+              }
+
+
+
+  ###
+  # Delete a specific app
+  #
+  delete:
```

Comment:
Made a deployable trait, which documents the force flag and added that to all related endpoints.